engine/src/main/java/io/zeebe/engine/state/mutable/MutableZeebeState.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.state.mutable;
+
+import io.zeebe.db.DbKey;
+import io.zeebe.db.DbValue;
+import io.zeebe.engine.state.KeyGenerator;
+import io.zeebe.engine.state.ZbColumnFamilies;
+import io.zeebe.engine.state.ZeebeState;
+import java.util.function.BiConsumer;
+
+public interface MutableZeebeState extends ZeebeState {
+
+  @Override
+  MutableDeploymentState getDeploymentState();
+
+  @Override
+  MutableWorkflowState getWorkflowState();
+
+  @Override
+  MutableJobState getJobState();
+
+  @Override
+  MutableMessageState getMessageState();
+
+  @Override
+  MutableMessageSubscriptionState getMessageSubscriptionState();
+
+  @Override
+  MutableMessageStartEventSubscriptionState getMessageStartEventSubscriptionState();
+
+  @Override
+  MutableWorkflowInstanceSubscriptionState getWorkflowInstanceSubscriptionState();
+
+  @Override
+  MutableIncidentState getIncidentState();
+
+  @Override
+  MutableBlackListState getBlackListState();
+
+  @Override
+  MutableVariableState getVariableState();
+
+  @Override
+  MutableTimerInstanceState getTimerState();
+
+  @Override
+  MutableElementInstanceState getElementInstanceState();
+
+  @Override
+  MutableEventScopeInstanceState getEventScopeInstanceState();
+
+  KeyGenerator getKeyGenerator();
+
+  /**
+   * Iterates over all entries for a given column family and presents each entry to the consumer.
+   *
+   * <p><strong>Hint</strong> Should only be used in tests.
+   *
+   * @param columnFamily the enum instance of the column family
+   * @param keyInstance this instance defines the type of the column family key type
+   * @param valueInstance this instance defines the type of the column family value type
+   * @param visitor the visitor that will be called for each entry
+   * @param <KeyType> the key type of the column family
+   * @param <ValueType> the value type of the column family
+   */
+  <KeyType extends DbKey, ValueType extends DbValue> void forEach(
+      final ZbColumnFamilies columnFamily,
+      final KeyType keyInstance,
+      final ValueType valueInstance,
+      final BiConsumer<KeyType, ValueType> visitor);
```

Comment:
Actually I didn't understand the implementation well enough. It calls createColumnFamily on zeebeDb and I'm not sure whether that counts as changing state. Also there's a comment stating:
>should only be used in tests

What should we do with this? In my opinion, the best case is that it would be removed from the interface altogether, but I don't think that's trivial. What do you think?

Reply:
I think we could move this into the test where it used. There we have probably also access to zeebeDb which is used by the impl

Reply:
@Zelldon Was relatively easy, please have a look :)